internal/provider/vaultradar/radar_source.go

```diff
@@ -0,0 +1,191 @@
+package vaultradar
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+	"github.com/hashicorp/terraform-provider-hcp/internal/provider/modifiers"
+
+	service "github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-radar/preview/2023-05-01/client/data_source_registration_service"
+)
+
+var (
+	_ resource.Resource              = &radarSourceResource{}
+	_ resource.ResourceWithConfigure = &radarSourceResource{}
+)
+
+// radarSourceResource is an implementation for configuring specific types Radar data sources.
+// Examples: hcp_vault_radar_source_github_cloud and hcp__vault_radar_source_github_enterprise make use of
```

Comment:
nit: Should be one underscore instead of the two  `hcp__vault_radar_source_github_enterprise` ?